File: go/clock/clock.go
package clock

import (
	"fmt"
	//"strconv"
)

type Clock struct {
	hour int
	minute int
}

const t = 24 * 60

func New(hour, minute int) Clock {
	hour = (hour + minute/60) % 24
	minute = minute % 60

	if hour < 0 {
		hour += 24
	}

	if minute < 0 {
		hour -= 1
		if hour < 0 {
			hour += 24
		}
		minute += 60
	}

	return Clock{hour, minute}
}

// Add func
func (c Clock) Add(m int) Clock{
	return New(c.hour + (c.minute + m)/60,(c.minute + m) % 60)
}

// Subtract func
func (c Clock) Subtract(m int) Clock{
	return New(c.hour + (c.minute - m)/60, (c.minute - m)% 60)
}

// String func
func (c Clock) String() string{
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

File: python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self.hour = (hour + minute // 60) % 24
        self.minute = minute % 60

    def __repr__(self):
        return "%02d:%02d" % (self.hour % 24, self.minute % 60)

    def __eq__(self, other):
        if other.hour == self.hour and other.minute == self.minute:
            return True
        return False

    def __add__(self, minutes):
        self.hour = self.hour + (self.minute + minutes) // 60
        self.minute = (self.minute + minutes) % 60
        return "%02d:%02d" % (self.hour % 24, self.minute % 60)

    def __sub__(self, minutes):
        if minutes <= 60:
            if minutes <= self.minute:
                self.minute = self.minute - minutes
            else:
                self.minute = self.minute - minutes + 60
                self.hour -= 1
        else:
            self.hour += (self.minute - minutes) // 60
            self.minute = (self.minute - minutes) % 60
        self.hour += 24 if self.hour < 0 else 0
        return "%02d:%02d" % (self.hour % 24, self.minute % 60)
